File: app/src/main/java/com/app/syspoint/db/dao/RolesDao.java
package com.app.syspoint.db.dao;

import com.app.syspoint.db.bean.RolesBean;
import com.app.syspoint.db.bean.RolesBeanDao;

import java.util.List;

public class RolesDao extends Dao {
    public RolesDao() {
        super("RolesBean");
    }

    final public List<RolesBean> getListaRolesByEmpleado(String identificador) {
        return dao.queryBuilder()
                .where(RolesBeanDao.Properties.Identificador.eq(identificador))

                .list();
    }




    //Retorna el empleado por identificador
    public final RolesBean getRolByEmpleado(String identificador, String modulo){
        final List<RolesBean> clienteBeans = dao.queryBuilder()
                .where(RolesBeanDao.Properties.Identificador.eq(identificador), RolesBeanDao.Properties.Modulo.eq(modulo))
                .list();
        return clienteBeans.size()> 0? clienteBeans.get(0) : null;
    }


    public final RolesBean getRolByModule(String identificador, String modulo){
        final List<RolesBean> clienteBeans = dao.queryBuilder()
                .where(RolesBeanDao.Properties.Identificador.eq(identificador), RolesBeanDao.Properties.Modulo.eq(modulo))
                .list();
        return clienteBeans.size()> 0? clienteBeans.get(0) : null;
    }

}

File: app/src/main/java/com/app/syspoint/repository/database/dao/RolesDao.kt
package com.app.syspoint.repository.database.dao

import com.app.syspoint.repository.database.bean.RolesBean
import com.app.syspoint.repository.database.bean.RolesBeanDao

class RolesDao: Dao("RolesBean") {

    fun getListaRolesByEmpleado(identificador: String?): List<RolesBean> {
        val roles =  dao.queryBuilder()
            .where(RolesBeanDao.Properties.Identificador.eq(identificador))
            .list() as List<RolesBean>
        return  roles
    }

    //Retorna el empleado por identificador
    fun getRolByEmpleado(identificador: String?, modulo: String?): RolesBean? {
        val clienteBeans = dao.queryBuilder()
            .where(
                RolesBeanDao.Properties.Identificador.eq(identificador),
                RolesBeanDao.Properties.Modulo.eq(modulo)
            )
            .list() as List<RolesBean>
        return if (clienteBeans.size > 0) clienteBeans[0] else null
    }

    fun getRolByModule(identificador: String?, modulo: String?): RolesBean? {
        val clienteBeans = dao.queryBuilder()
            .where(
                RolesBeanDao.Properties.Identificador.eq(identificador),
                RolesBeanDao.Properties.Modulo.eq(modulo)
            )
            .list() as List<RolesBean>
        return if (clienteBeans.size > 0) clienteBeans[0] else null
    }
}